elki/src/main/java/de/lmu/ifi/dbs/elki/utils/containers/RankStruct.java

```diff
@@ -0,0 +1,15 @@
+package de.lmu.ifi.dbs.elki.utils.containers;
+
+/**
+ * Structure to hold return values in index creation for MCDEDependenceEstimate
+ */
+
+public class RankStruct {
+    public int index;
```

Comment:
This is essentially a boxed integer - a fairly costly object due to Java object memory management. We usually avoid small objects that occur in large quantities because of these overheads.